src/test/java/seedu/address/logic/commands/EditPropertyCommandTest.java

```diff
@@ -0,0 +1,177 @@
+                .withPostalCode(VALID_POSTAL_BURGHLEY_DRIVE).withType(VALID_TYPE_BURGHLEY_DRIVE).build();
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(indexLastProperty, descriptor);
+
+        String expectedMessage = String.format(EditPropertyCommand.MESSAGE_SUCCESS, editedProperty);
+
+        Model expectedModel = new ModelManager(new PropertyBook(model.getPropertyBook()), new UserPrefs());
+        expectedModel.setProperty(lastProperty, editedProperty);
+
+        assertCommandSuccess(editPropertyCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_noFieldSpecifiedUnfilteredList_success() {
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(INDEX_FIRST_PROPERTY,
+                new EditPropertyDescriptor());
+        Property editedProperty = model.getFilteredPropertyList().get(INDEX_FIRST_PROPERTY.getZeroBased());
+
+        String expectedMessage = String.format(EditPropertyCommand.MESSAGE_SUCCESS, editedProperty);
+
+        Model expectedModel = new ModelManager(new PropertyBook(model.getPropertyBook()), new UserPrefs());
+
+        assertCommandSuccess(editPropertyCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_filteredList_success() {
+        showPropertyAtIndex(model, INDEX_FIRST_PROPERTY);
+
+        Property propertyInFilteredList = model.getFilteredPropertyList().get(INDEX_FIRST_PROPERTY.getZeroBased());
+        Property editedProperty = new PropertyBuilder(propertyInFilteredList)
+                .withName(VALID_NAME_BURGHLEY_DRIVE).build();
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(INDEX_FIRST_PROPERTY,
+                new EditPropertyDescriptorBuilder().withName(VALID_NAME_BURGHLEY_DRIVE).build());
+
+        String expectedMessage = String.format(EditPropertyCommand.MESSAGE_SUCCESS, editedProperty);
+
+        Model expectedModel = new ModelManager(new PropertyBook(model.getPropertyBook()), new UserPrefs());
+        expectedModel.setProperty(model.getFilteredPropertyList().get(0), editedProperty);
+
+        assertCommandSuccess(editPropertyCommand, model, expectedMessage, expectedModel);
+    }
+
+    @Test
+    public void execute_duplicatePropertyUnfilteredList_failure() {
+        Property firstProperty = model.getFilteredPropertyList().get(INDEX_FIRST_PROPERTY.getZeroBased());
+        EditPropertyDescriptor descriptor = new EditPropertyDescriptorBuilder(firstProperty).build();
+        EditPropertyCommand editPropertyCommand = new EditPropertyCommand(INDEX_SECOND_PROPERTY, descriptor);
+
+        assertCommandFailure(editPropertyCommand, model, EditPropertyCommand.MESSAGE_DUPLICATE_PROPERTY);
+    }
+
+    @Test
+    public void execute_duplicatePropertyFilteredList_failure() {
+        showPropertyAtIndex(model, INDEX_FIRST_PROPERTY);
+
+        // edit Property in filtered list into a duplicate in address book
```

Comment:
Should it be property book instead of address book here?

Reply:
You have a good eye! should be fixed!